web/package.json

```diff
@@ -48,6 +48,7 @@
   "dependencies": {
     "@types/node": "^11.9.4",
     "es6-shim": "^0.35.5",
+    "events": "^3.1.0",
```

Comment:
One of the original design goals for KeymanWeb was zero external dependencies (this is pre-npm, remember). We may the explicit decision to not use jquery because (a) it added substantially to the size of the final js, and (b) there was potential for compat issues between different libraries, esp those that registered globals or extended standard objects.

I see that the unpacked size of events is 73kb -- quite large. That's substantially larger than KeymanWeb used to be altogether.

Is it really worth importing this library?

Reply:
> I see that the unpacked size of events is 73kb -- quite large. That's substantially larger than KeymanWeb used to be altogether.

The linked JS file itself, before minification, is 13kb.  Definitely not _nothing_, but it's at least fairly light.

<img width="882" alt="Screen Shot 2020-04-16 at 7 57 38 AM" src="https://user-images.githubusercontent.com/25213402/79403049-f981d680-7fb7-11ea-952c-4e70a315d3b6.png">

That said, I just found a smaller-scale one that's only 9 kb before minification - even smaller - that gets around the `module` problem noted elsewhere within the comments.  So... pardon as I convert to that.

> Is it really worth importing this library?

This is the real question, I think.  We don't _have_ to use it, since KMW does have some pre-existing event handling code, though using that here actually has its own costs.

`kmwutils.ts` defines the core `addEventListener`, `removeEventListener`, etc implementations currently used throughout KMW.
- This file is quite DOM-aware.
- So, to use its event-handling functionality in web-core, I'd have to refactor our existing event-handling code.

Admittedly, the code within kmwutils.ts for event handling has been unchanging so far, though it has had some associated costs.
- It's up to the objects that _register_ events to create events with prefixes that ensure uniqueness.
- Event handling isn't done in a properly object-oriented way at present within KMW.
    - Events are (generally) registered against `string` tags, not the object that generates the event.
    - With the `modelManager` case above... that's simply an alias that manages the tag,  giving the illusion that it is object oriented.
- Furthermore, it relies upon the existence of a singleton object - `keyman.util` - to store event management data.
    - This is fairly bad for unit testing, as the object must be force-purged between unit tests... and because web-core doesn't yet _have_ any such singletons to utilize.
    - One big 'plus' of `EventEmitter` is that it includes a `removeAllListeners` method, which is fantastic for cleanup operations - especially since it cleans up the listeners on just that one object.

Basically, if I were to modernize/properly-refactor our existing event management code, `EventEmitter` is how I would want to do it.  It's the JS community standard these days, it's object-oriented... and it's externally maintained.

Even if we wanted to use the kmwutils.ts code, it'd require some refactoring... and some of the design questions (getting around that singleton) aren't that easy, especially if we want to keep things clean and pretty.

> One of the original design goals for KeymanWeb was zero external dependencies (this is pre-npm, remember).

That makes sense, and I generally prefer it that way.  The less dependencies - especially ones that tend to require a lot of their own - the better.  For what it's worth, neither the `events` package, nor the `eventemitter3` package I'm about to replace it with, have any dependencies.  They're small-scale and focused... especially the latter.